From 37a9e657e84b20ca1f8371589be1cf52d5b16cc5 Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Wed, 30 Nov 2011 12:26:47 -0700
Subject: [PATCH] Move review button off preferences into title bar

The review button did not fit well on the preferences
panel.  It was not a preference and it operated
immediately without having to hit Update or Save.

To encourage its usage more, move it to the title
bar on the far right of the screen above the link
to the next file.  This makes it much more accessible
since it will now always be on the screen and it is
positioned nicely near the action a user will most
likely take next after checking the reviewed box.
This last piece should make reviewing on smaller
screens such as tablets much more convenient.

Change-Id: Ia6d47cc8b976805297231916dc28f7d8002fbaf4
---
 .../google/gerrit/client/patches/PatchScreen.java  |   25 +++++++++++++------
 .../client/patches/PatchScriptSettingsPanel.java   |    8 ------
 .../client/patches/PatchScriptSettingsPanel.ui.xml |    9 -------
 3 files changed, 17 insertions(+), 25 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScreen.java
index 910bf61..b165b1e 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScreen.java
@@ -39,7 +39,9 @@ import com.google.gwt.event.logical.shared.ValueChangeEvent;
 import com.google.gwt.event.logical.shared.ValueChangeHandler;
 import com.google.gwt.event.shared.HandlerRegistration;
 import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.CheckBox;
 import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.InlineLabel;
 import com.google.gwt.user.client.ui.Label;
 import com.google.gwtexpui.globalkey.client.GlobalKey;
 import com.google.gwtexpui.globalkey.client.KeyCommand;
@@ -105,6 +107,7 @@ public abstract class PatchScreen extends Screen implements
   protected PatchScriptSettingsPanel settingsPanel;
   protected TopView topView;
 
+  private CheckBox reviewed;
   private HistoryTable historyTable;
   private FlowPanel topPanel;
   private FlowPanel contentPanel;
@@ -151,6 +154,15 @@ public abstract class PatchScreen extends Screen implements
     idSideB = diffSideB != null ? diffSideB : id.getParentKey();
     this.patchIndex = patchIndex;
 
+    reviewed = new CheckBox(Util.C.reviewed());
+    reviewed.addValueChangeHandler(
+        new ValueChangeHandler<Boolean>() {
+          @Override
+          public void onValueChange(ValueChangeEvent<Boolean> event) {
+            setReviewedByCurrentUser(event.getValue());
+          }
+        });
+
     prefs = fileList != null ? fileList.getPreferences() :
                                new ListenableAccountDiffPreference();
     if (Gerrit.isSignedIn()) {
@@ -165,13 +177,6 @@ public abstract class PatchScreen extends Screen implements
         });
 
     settingsPanel = new PatchScriptSettingsPanel(prefs);
-    settingsPanel.getReviewedCheckBox().addValueChangeHandler(
-        new ValueChangeHandler<Boolean>() {
-          @Override
-          public void onValueChange(ValueChangeEvent<Boolean> event) {
-            setReviewedByCurrentUser(event.getValue());
-          }
-        });
   }
 
   @Override
@@ -234,6 +239,10 @@ public abstract class PatchScreen extends Screen implements
   protected void onInitUI() {
     super.onInitUI();
 
+    if (Gerrit.isSignedIn()) {
+      setTitleFarEast(reviewed);
+    }
+
     keysNavigation = new KeyCommandSet(Gerrit.C.sectionNavigation());
     keysNavigation.add(new UpToChangeCommand(patchKey.getParentKey(), 0, 'u'));
     keysNavigation.add(new FileListCmd(0, 'f', PatchUtil.C.fileList()));
@@ -444,7 +453,7 @@ public abstract class PatchScreen extends Screen implements
 
     // Mark this file reviewed as soon we display the diff screen
     if (Gerrit.isSignedIn() && isFirst) {
-      settingsPanel.getReviewedCheckBox().setValue(true);
+      reviewed.setValue(true);
       setReviewedByCurrentUser(true /* reviewed */);
     }
 
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.java
index 70aade1..df0fff5 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.java
@@ -79,9 +79,6 @@ public class PatchScriptSettingsPanel extends Composite implements
   CheckBox showTabs;
 
   @UiField
-  CheckBox reviewed;
-
-  @UiField
   CheckBox skipDeleted;
 
   @UiField
@@ -119,7 +116,6 @@ public class PatchScriptSettingsPanel extends Composite implements
     initIgnoreWhitespace(ignoreWhitespace);
     initContext(context);
     if (!Gerrit.isSignedIn()) {
-      reviewed.setVisible(false);
       save.setVisible(false);
     }
 
@@ -188,10 +184,6 @@ public class PatchScriptSettingsPanel extends Composite implements
     syntaxHighlighting.setTitle(title);
   }
 
-  public CheckBox getReviewedCheckBox() {
-    return reviewed;
-  }
-
   public AccountDiffPreference getValue() {
     return listenablePrefs.get();
   }
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.ui.xml b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.ui.xml
index f320ce5..e819ffa 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.ui.xml
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/patches/PatchScriptSettingsPanel.ui.xml
@@ -164,15 +164,6 @@ limitations under the License.
         <ui:attribute name='text'/>
       </g:Button>
     </td>
-
-    <td>
-      <g:CheckBox
-          ui:field='reviewed'
-          text='Reviewed'
-          tabIndex='15'>
-        <ui:attribute name='text'/>
-      </g:CheckBox>
-    </td>
   </tr>
 
   <tr valign='top'>
-- 
1.7.6

